mk/config.mk

```diff
@@ -35,11 +35,15 @@ WARNS ?= 3
 # Define NOWERROR=1 so that warnings are not treated as errors
 # NOWERROR=1
 
-# Define DEBUG=1 to compile with -g option
+# Define DEBUG=1 to compile without optimization (forces -O0)
 # DEBUG=1
 
-# If 1, debug mode of the tee firmware (CPU restart, Core Status)
-CFG_TEE_CORE_DEBUG ?= 0
+# If y, enable debug mode of the tee firmware (CPU restart, Core Status
+# verbose, panic & assert verbose). When disable, NDEBUG directive is defined.
+CFG_TEE_CORE_DEBUG ?= n
+ifneq ($(CFG_TEE_CORE_DEBUG),y)
+NDEBUG ?= 1
```

Comment:
This will affect user TAs too, I'm not sure that a `CFG_TEE_CORE_*` flag should impact user TAs directly.